ansible/roles/test/files/ptftests/arista.py

```diff
@@ -288,7 +288,25 @@ def parse_bgp_neighbor_once(self, output):
 
         return is_gr_ipv4_enabled, is_gr_ipv6_enabled, restart_time
 
-    def parse_bgp_neighbor(self, output):
+    def parse_bgp_neighbor(self, output, get_neigh=False, get_asn=False):
+        # ASN info is always retrieved along with neighbor info
+        if get_neigh or get_asn:
+            neigh_bgp = None
+            dut_bgp = None
+            asn = None
+            for line in output.split('\n'):
+                if 'BGP neighbor is' in line:
+                   dut_bgp = re.findall('BGP neighbor is (.*?),', line)[0]
+                elif 'Local AS is' in line and get_asn:
+                   asn = re.findall('Local AS is (\d+?),', line)[0]
+                elif 'Local TCP address is' in line:
+                   neigh_bgp = re.findall('Local TCP address is (.*?),', line)[0]
+                   break
+
+            if get_asn:
+               return neigh_bgp, dut_bgp, asn
+            return neigh_bgp, dut_bgp
```

Comment:
The first part could be extracted as a standalone function. 
The semantics is strange, there are multiple bgp neighbor IP addresses, why return only one? The parameters are not independent with each other, I suggest you return them all without input parameters at all.

Reply:
Will move that to a standalone function and remove the input parameters